cmake/Modules/CocosBuildHelpers.cmake

```diff
@@ -33,14 +57,104 @@ function(cocos_mark_resources)
             MACOSX_PACKAGE_LOCATION "${opt_RESOURCEBASE}/${RES_LOC}"
             HEADER_FILE_ONLY 1
             )
+
+        if(XCODE OR VS)
+            string(REPLACE "/" "\\" ide_source_group "${opt_RESOURCEBASE}/${RES_LOC}")
+            source_group("${ide_source_group}" FILES ${RES_FILE})
+        endif()
     endforeach()
 endfunction()
 
+# mark the files in the sub dir of CMAKE_CURRENT_SOURCE_DIR
+function(cocos_mark_code_files cocos_target)
+  set(oneValueArgs GROUPBASE)
+  cmake_parse_arguments(opt "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+  if(NOT opt_GROUPBASE)
+    set(root_dir ${CMAKE_CURRENT_SOURCE_DIR})
+  else()
+    set(root_dir ${opt_GROUPBASE})
+    message(STATUS "target ${cocos_target} code group base is: ${root_dir}")
+  endif()
+
+  message(STATUS "cocos_mark_code_files: ${cocos_target}")
+  set(group_base "Source Files")
+
+  get_property(file_list TARGET ${cocos_target} PROPERTY SOURCES)
+
+  foreach(single_file ${file_list})
+    # get relative_path
+    get_filename_component(abs_path ${single_file} ABSOLUTE)
+    file(RELATIVE_PATH relative_path_with_name ${root_dir} ${abs_path})
+    get_filename_component(relative_path ${relative_path_with_name} PATH)
+    # set source_group, consider sub source group 
+    string(REPLACE "/" "\\" ide_file_group "${group_base}/${relative_path}")
+    source_group("${ide_file_group}" FILES ${single_file})
+
+  endforeach()
+  
+endfunction()
+
+# if cc_variable not set, then set it cc_value
+macro(cocos_fake_set cc_variable cc_value)
+  if(NOT DEFINED ${cc_variable})
+    set(${cc_variable} ${cc_value})
+  endif()
+endmacro()
+
+# macos package, need review
+macro(cocos_pak_xcode cocos_target)
```

Comment:
There is a info.plist file, why need to set it here?

Reply:
it's a info.plist file template, it will become the macOS bundle info.plist

Reply:
this macro set it as default value, if you don't want the default value, just use like:
```cmake
    if(MACOSX)
        cocos_pak_xcode(${APP_NAME} BUNDLE_NAME "Cpp Empty Test")
    elseif(IOS)
        cocos_pak_xcode(${APP_NAME} INFO_PLIST "iOSBundleInfo.plist.in")
    endif()
```

Reply:
I mean, why not use exist info.plist file? Developers uses to add a plist.info in project.

Reply:
understood, use the plist.info in project is a better choice, I will modify it

Reply:
https://github.com/cocos2d/cocos2d-x/blob/f1dbc310ba6e7743b33e70354739270138bad4be/tests/cpp-empty-test/proj.mac/Info.plist#L8

variables  like `EXECUTABLE_NAME ` depend on the Xcode package config, CMake didn't support this, the same meaning variable of `EXECUTABLE_NAME` in CMake is `MACOSX_BUNDLE_EXECUTABLE_NAME` if we use info.plist template. 

So if we use plist.info in project resource, I am afraid that we have to change those variables into value to avoid depend Xcode config.

Reply:
another information is that when set `info.plist` in project into Xcode resource, the variables which cmake didn't support disappeared, meanwhile the open source project like `Urho3D` use info.plist template

Reply:
understood that, I will modify the variables in info.plist to values, so that cocos-console need modify to adapt this, when using `cocos new`, after all cmake works finished.